RadixWallet/Features/ProfileBackupsFeature/RestoreProfileFromBackup/Children/ImportSeedPhrasesFlow/ImportMnemonicsFlowCoordinator.swift

```diff
@@ -255,20 +255,19 @@ public struct ImportMnemonicsFlowCoordinator: Sendable, FeatureReducer {
 			return .none
 		} else {
 			state.destination = nil
-			guard let newMainBDFS = state.newMainBDFS else {
-				return .send(.delegate(.finishedImportingMnemonics(skipped: state.skipped, imported: state.imported, notYetSavedNewMainBDFS: nil)))
-			}
-			if state.context.isFromOnboarding {
-				return .send(.delegate(.finishedImportingMnemonics(skipped: state.skipped, imported: state.imported, notYetSavedNewMainBDFS: state.newMainBDFS)))
-			} else {
-				return .run { [skipped = state.skipped, imported = state.imported] send in
+			return .run { [skipped = state.skipped, imported = state.imported, newMainBDFS = state.newMainBDFS, context = state.context] send in
+				if let newMainBDFS, !context.isFromOnboarding {
 					try await factorSourcesClient.saveNewMainBDFS(newMainBDFS)
-					return await send(.delegate(.finishedImportingMnemonics(
-						skipped: skipped,
-						imported: imported,
-						notYetSavedNewMainBDFS: nil
-					)))
 				}
+
+				/// A small delay is needed after dismissal in order to not break Home screen modal presentations
+				try? await Task.sleep(for: .milliseconds(100))
```

Comment:
Maybe use 250ms or something, I think we used 100 ms in the past and it was not enough on some devices.. this might ofc have changed with iOS updates. But in general I think best be on the safe side when using delay based solutions.